nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -449,3 +451,26 @@ def _set_weight_compression_config(
             weight_param.compression_config = primary_config
     else:
         _assign_mixed_precision(internal_weight_params, ratio, primary_config)
+
+
+def create_constant_node(data: np.ndarray, dtype: ov.Type, name: str) -> ov.op.Constant:
+    """
+    Creates a constant node with the given constant value, data type, and name.
+
+    :param data: The constant value.
+    :param dtype: The data type of the constant node.
+    :param name: The name to be assigned to the constant node.
+    :return: The OpenVINO constant node.
+    """
+    if dtype in [ov.Type.i4, ov.Type.u4]:
```

Comment:
I would implement i4 and u4 the same way as it is done for NF4. I consider this as a problem on OpenVINO side. So we can easily switch to opset13 once it is fixed.
In a nutshell, my motivation is the following: don't do the packing on your side because it is up to OpenVINO how to pack data types internally.